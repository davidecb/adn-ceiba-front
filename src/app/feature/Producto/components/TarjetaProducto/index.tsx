import * as PropTypes from 'prop-types';
import * as React from 'react';
import {
  CostoProducto,
  NombreProducto,
  ProductoImagen,
  TarjetaProductoContainer,
  TiempoEjecucion
} from './styles';
import { BtnPersonalizarProducto } from '../BtnPersonalizarProducto';
import { Producto } from '../../models/Producto';


export interface TarjetaProductoProps {
  producto: Producto;
  onClickPersonalizarProducto: (producto: Producto) => void;
}

export const TarjetaProducto: React.FC<TarjetaProductoProps> = ({
  producto,
  onClickPersonalizarProducto,
}) => {
  return (
    <TarjetaProductoContainer>
      <ProductoImagen src={producto.imagen} />
      <NombreProducto>{producto.nombre}</NombreProducto>
      <CostoProducto>$ {producto.costo}</CostoProducto>
      <TiempoEjecucion>{producto.tiempo} min.</TiempoEjecucion>
      <BtnPersonalizarProducto onPersonalizar={onClickPersonalizarProducto} producto={producto} />
    </TarjetaProductoContainer>
  );
};

TarjetaProducto.propTypes = {
  producto: PropTypes.any.isRequired,
  onClickPersonalizarProducto: PropTypes.func.isRequired,
};
