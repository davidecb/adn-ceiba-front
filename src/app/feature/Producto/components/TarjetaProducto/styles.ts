import styled from 'styled-components';

export const TarjetaProductoContainer = styled.div`
  width: 250px;
  height: 320px;
  margin: 10px;
  padding: 10px;
  border: 2px groove rgb(10, 10, 114);
  border-radius: 10px;
  display: flex;
  flex-flow: column;
  align-items: center;
  justify-content: center;
`;

export const NombreProducto = styled.div`
  width: 100%;
  font-size: large;
  text-align: center;
  font-weight: bolder;
`;

export const CostoProducto = styled.div`
  width: 100%;
  font-size: large;
  text-align: center;
`;

export const TiempoEjecucion = styled.div`
  width: 100%;
  font-size: large;
  text-align: center;
`;

export const ProductoImagen = styled.img`
  border: 3px outset rgb(200, 200, 200);
  border-radius: 10px;
  min-height: 180px;
  min-width: 200px;
  max-height: 180px;
  max-width: 200px;
  object-fit: contain;
`;
