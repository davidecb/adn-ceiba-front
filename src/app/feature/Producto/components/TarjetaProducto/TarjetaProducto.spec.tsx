import * as React from 'react';
import { RenderResult, render } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { TarjetaProducto } from './../TarjetaProducto';

describe('TarjetaProducto Test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof TarjetaProducto> & {
    onClickPersonalizarProducto: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      producto: {
        id: 1,
        nombre: 'producto testing',
        costo: 10000,
        tiempo: 30,
        imagen: 'imagen.jpg'
      },
      onClickPersonalizarProducto: stub(),
    };
    componentWrapper = render(<TarjetaProducto {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });
});

export {};
