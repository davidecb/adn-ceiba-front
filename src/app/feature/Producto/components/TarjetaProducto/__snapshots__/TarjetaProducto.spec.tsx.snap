// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TarjetaProducto Test should match snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm dERIyq"
  >
    <img
      class="sc-AxhUy YCAwq"
      src="imagen.jpg"
    />
    <div
      class="sc-AxirZ cNXfXq"
    >
      producto testing
    </div>
    <div
      class="sc-AxiKw hjBCiN"
    >
      $ 
      10000
    </div>
    <div
      class="sc-AxhCb jwgvDN"
    >
      30
       min.
    </div>
    <button
      class="sc-AxgMl cMdgkA"
    >
      <span>
        personalizar
      </span>
    </button>
  </div>
</div>
`;
